### DISCUSSION TITLE
Establishing IKE_SA failed, peer not responding

### DISCUSSION THREAD

**Discussion by yoloknight** [strongswan/strongswan#371] [Q&A]
Hi guys,

I´m currently working on a IPSec project and want to connect a Fortigate with my Laptop (Roadwarrior). The communication between these two guys should support IPv4 and IPv6 communication. The sad thing is, my configuration dont work. Can you tell me what I´m doing wrong?

I have tryed a combination of this config: https://www.strongswan.org/testing/testresults/swanctl/rw-cert/
and this one: https://wiki.strongswan.org/attachments/2096/split-tunnel-ikev2.txt

Here my swanctl config:

` connections {

     ipsec {

         version = 2

         remote_addrs = 20.20.20.8

         proposals = aes256-sha256-modp2048

         vips = 0.0.0.0,::

         encap = yes

         mobike = no

         dpd_delay = 30s

         dpd_timeout = 150s

         fragmentation = no

         unique = no

         rekey_time = 1800s

         local-1 {

             certs = Example_1.crt

             auth = psk

             id =%any
         }

         remote {

             id = %any

             certs = Fortigate_Gateway.crt 

             auth = psk
         }

         children {

             ipsec6 {

                 esp_proposals = aes256-sha256-modp2048

                 remote_ts = ::/0

                 start_action = start

                 close_action = start

             }

         }
        
         children {

             ipsec4 {

                 esp_proposals = aes256-sha256-modp2048

                 remote_ts = 0.0.0.0/0

                 start_action = start

                 close_action = start

             }

         }

     }

 }

 secrets {
 
     ike-ipsec {

         secret = test
     }
 }

 authorities {

     auth-2 {

         cacert = The_CA.crt
     }
 }
`

Here the message I got:

[IKE] initiating IKE_SA ipsec[12] to 20.20.20.8
[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(HASH_ALG) N(REDIR_SUP) ]
[NET] sending packet: from 192.168.1.5[500] to 20.20.20.8[500] (456 bytes)
[NET] received packet: from 20.20.20.8[500] to 192.168.1.5[500] (441 bytes)
[ENC] parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ ]
[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
[IKE] local host is behind NAT, sending keep alives
[IKE] received 1 cert requests for an unknown ca
[CFG] no IDi configured, fall back on IP address
[IKE] authentication of '192.168.1.5' (myself) with pre-shared key
[IKE] establishing CHILD_SA ipsec4{8}
[ENC] generating IKE_AUTH request 1 [ IDi AUTH CPRQ(ADDR ADDR6 DNS DNS6) SA TSi TSr N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
[NET] sending packet: from 192.168.1.5[4500] to 20.20.20.8[4500] (304 bytes)
[IKE] retransmit 1 of request with message ID 1
[NET] sending packet: from 192.168.1.5[4500] to 20.20.20.8[4500] (304 bytes)
[IKE] retransmit 2 of request with message ID 1
[NET] sending packet: from 192.168.1.5[4500] to 20.20.20.8[4500] (304 bytes)
[IKE] retransmit 3 of request with message ID 1
[NET] sending packet: from 192.168.1.5[4500] to 20.20.20.8[4500] (304 bytes)
[IKE] sending keep alive to 20.20.20.8[4500]
[IKE] retransmit 4 of request with message ID 1
[NET] sending packet: from 192.168.1.5[4500] to 20.20.20.8[4500] (304 bytes)
[IKE] sending keep alive to 20.20.20.8[4500]
[IKE] sending keep alive to 20.20.20.8[4500]
[IKE] retransmit 5 of request with message ID 1
[NET] sending packet: from 192.168.1.5[4500] to 20.20.20.8[4500] (304 bytes)
[IKE] sending keep alive to 20.20.20.8[4500]
[IKE] sending keep alive to 20.20.20.8[4500]
[IKE] sending keep alive to 20.20.20.8[4500]
[IKE] giving up after 5 retransmits
[IKE] establishing IKE_SA failed, peer not responding
initiate failed: establishing CHILD_SA 'ipsec4' failed

**Comment by tobiasbrunner** [comment]
The peer does not respond to the IKE_AUTH message. Either it doesn't receive it (e.g. because UDP port 4500 is blocked by some firewall/router) or it doesn't like it (it should respond with an error, though). So check your firewall settings and the log of the peer.

Also, why do you have certificates configured if you authenticate via PSK?

**Reply by yoloknight** [reply]
Hi Tobias,
well that with PSK was a mistake, I experimented with the config with pubkey and psk. In the orignal config I have "pubkey" like the example for cert communication.

In my fortigate I get this: 
`id=20085 trace_id=2 func=print_pkt_detail line=5617 msg="vd-root:0 received a packet(proto=17, 80.187.100.163:9456->20.20.20.8:4500) from wan1. "
id=20085 trace_id=2 func=resolve_ip_tuple_fast line=5697 msg="Find an existing session, id-00423e12, original direction"`

Also in the logs there is no block of policy deny. I had in the past tried this configuration with the network manager, but the network manager cant handle 2 child_sa from the Fortigate. (In a other Ticket Tobias we both talked about it :) )

For me the error seems that he cant handle the Gateway.crt correctly. Is my guess here perhaps right?

**Reply by tobiasbrunner** [reply]
At least it seems to receive the packet. Is nothing more logged? Maybe it doesn't like it (e.g. because it uses the wrong authentication method) but doesn't respond (would be strange), or maybe it does respond but there is an IP fragmentation issue because the server certificate is too big. Why did you disable IKE fragmentation explicitly? Does the Fortigate box not support this (maybe has to be enabled)?

**Reply by yoloknight** [reply]
Well the  authentification method on the Fortigate is "Signature" (Gateway.crt)
Fragmentation was deactivated, because it makes no diffrence in the outcome. Also I had experimented in the past with the network-manager he had also a problem with the encapsulation. Without it I got a connection from Strongswan to the Fortigate but only with IPv4 communication, not with both (IPv6). 

The Fortigate give me only this looks you already have seen. (diag debug flow port 4500; diag debug flow trace start)

I give you my test setup form the Fortigate:

config vpn ipsec phase1-interface

    edit "test_ipsec"

        set type dynamic

        set interface "wan1"

        set ike-version 2

        set authmethod signature

        set peertype peergrp

        set net-device enable

        set mode-cfg enable

        set proposal aes128-sha256 aes256-sha256 3des-sha256 aes128-sha1 aes256-sha1 3des-sha1

        set dpd on-idle

        set nattraversal forced

        set certificate "Gateway"   (means Gateway.crt)

        set peergrp "Test_Dialin"

        set ipv4-start-ip 192.168.1.100

        set ipv4-end-ip 192.168.1.110

        set ipv4-netmask 255.255.255.0

        set ipv6-start-ip 2002:5555:5555:5555::100 (fake ipv6, not the real ones)

        set ipv6-end-ip 2002:5555:5555:5555::199 (fake ipv6, not the real ones)

        set ipv6-prefix 64

    next

end


config vpn ipsec phase2-interface

    edit "IPv4"

        set phase1name "Test_IPSec"

        set proposal aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm chacha20poly1305

    next

    edit "IPv6"

        set phase1name "Test_IPSec"

        set proposal aes128-sha1 aes256-sha1 aes128-sha256 aes256-sha256 aes128gcm aes256gcm chacha20poly1305

        set src-addr-type subnet6

        set dst-addr-type subnet6

    next

end

This setup worked before with the network-manager strongswan plugin.

**Reply by tobiasbrunner** [reply]
Then your config is obviously wrong. Use `auth = pubkey`.

**Reply by yoloknight** [reply]
Hi Tobias,

I fixed my configuration already to pubkey, but the error was not in the config itself. It seems that the certificate was not valid. The strange thing is, that in Strongswan + Networkmanager these certificates where valid. So I made new certificates and I have a connection to my Fortigate. And so far it seems the two child_sa works.

For someone who wants to implement what I did. Do the following:
-> Copy or make a similar config as my Fortigate Config
-> Use the Roadwarrior example from Strongswan (https://www.strongswan.org/testing/testresults/swanctl/rw-cert/carol.swanctl.conf) -> then config a second child
-> check the shitty certificates, because these Ä'#!_ will make your life hard.

I think this topic can be closed. If I have new errors, then it will be a new topic. :)
Thanks for your help @tobiasbrunner :)

**Reply by yoloknight** [reply]
To clearify for some viewers, I use the CLI Swanctl here, not the Networkmanager. If you want to have IPv6 and IPv4 communication you have to use the cli, so what I did here. (or maybe somewhere in the future the Networkmanager + Strongswan accept Ipv6 and IPv4 communication)

**Reply by tobiasbrunner** [reply]
It already does, but you need relatively recent versions (at least 5.8.0 for strongSwan and 1.4.5 for the NM plugin, newer versions are recommended).
